warp/codegen.py

```
@@ -4107,6 +4152,10 @@ def codegen_func(adj, c_func_name: str, device="cpu", options=None):
     forward_args = []
     reverse_args = []
 
+    import traceback
+    traceback.print_stack()
+    print(f"FORWARD ARGS ... self.options={self.options}");
```

Ruff (0.14.2)

`self` may be undefined, or defined from star imports

(F405)